### DISCUSSION TITLE
Why do I see the admin panel with access denied?

### DISCUSSION THREAD

**Discussion by AlexKomMsk** [Laravel-Backpack/community-forum#245] [General] [answered]
In the config, I specified the middleware
`App\Http\Middleware\CheckIfAdmin::class,`

I try to go to the admin url with no admin user and see the edges of the admin panel on the background of the 404 page. 
![2022-10-11_14-08-15](https://user-images.githubusercontent.com/61763465/195077110-7aa7cd82-7581-43ab-8c03-bb20a0be1c9a.png)
Why do I see the left admin menu? How to turn it off?

**Comment by AlexKomMsk** [comment]
Removed check in file
`views.errors.layout`

It is better to think over this point otherwise there may be security problems in the admin area.

**Comment by pxpm** [comment, marked as answer]
@AlexKomMsk what is the contents of `App\Http\Middleware\CheckIfAdmin::class` ? 

If I return `false` from `checkIfUserIsAdmin()` I don't see any sidebar. 

If you are talking about `roles` and `permissions`, you should add your checks directly in the `sidebar` file:

```php
@if(backpack_user()->hasRole('admin'))
//show items only available for users with admin role
@endif
``` 

Also you should add those permission checks in your controllers so that even if someone try direct access to the resource url, the controller would deny the access. 
You can control that access with `$this->crud->allowAccess(['list','create'])` and `denyAccess->` in your controllers. 

Cheers
